packages/cli/src/commands/remove.ts

```diff
@@ -129,37 +135,38 @@ export default async function main(client) {
       .join(' ')} in ${chalk.bold(contextName)}`
   );
 
-  let aliases;
-  let projects;
-  let deployments;
+  let aliases: Alias[][];
+  let projects: Project[];
+  let deployments: DeploymentPartial[];
   const findStart = Date.now();
 
   try {
-    const searchFilter = d =>
-      ids.some(
-        id =>
-          d &&
-          !(d instanceof NowError) &&
-          (d.uid === id || d.name === id || d.url === normalizeURL(id))
-      );
-
     const [deploymentList, projectList] = await Promise.all([
       Promise.all(
         ids.map(idOrHost =>
-          getDeploymentByIdOrHost(client, contextName, idOrHost)
+          getDeploymentByIdOrHost(client, contextName!, idOrHost)
         )
       ),
       Promise.all(
         ids.map(async idOrName => getProjectByIdOrName(client, idOrName))
       ),
     ]);
 
-    deployments = deploymentList.filter(searchFilter);
-    projects = projectList.filter(searchFilter);
+    deployments = deploymentList
+      .filter((d): d is any => d && 'name' in d)
```

Comment:
@styfle @TooTallNate  The only way that could be working

Reply:
The previous solution `DeploymentPartial` no working. In this case, the "best solution" was to opt for any